### DISCUSSION TITLE
battery_sim_1.104

### DISCUSSION THREAD

**Discussion by hif2k1** [hif2k1/battery_sim#68] [Announcements]
**Please post the battery setups you are using so these can be added to the pre-configured options! Post an issue/comment on github.**

**In this update:**
Improve config flow
Add support for static tariffs
Fix problems with import and export meter tracking
Fix startup problem where battery not starting till restart of home assistant
Add battery cycle meter

<hr /><em>This discussion was created from the release <a href='https://github.com/hif2k1/battery_sim/releases/tag/battery_sim_1.104'>battery_sim_1.104</a>.</em>

**Comment by zaizuney** [comment]
Hi,
I'm running 3 VM's,
1) no battery
2) 1 tesla powerwall
3) 2 tesla powerwalls (however how to configure this is not so clear)

**Reply by hif2k1** [reply]
Thanks for the comment. To configure 2 Tesla Powerwalls you would have to create a custom battery with double a capacity of 27kwh, a max charging speed of 7.36 kw, a max discharging speed of 10kw and an efficiency of 1.8. However, we could make a template for this which is a good suggestion.

**Comment by erwano1957** [comment]
Hi,
I was looking for an extension that allows the simulation of a battery and came across your development. For my current use case it seems to fit ideally.
I'm trying to setup a virtual Battery (2.4 kWh) and connected it with the required sensors. All the sensors seem to be ok. When there is more production than consumption, the battery starts to charge and always exactly after 10 sec stops charging.
This leads to the fact that the majority of the excess energy is fed back into the grid.
What can I do to narrow down or even fix this behavior?
Thanks in advance for the support.

**Reply by hif2k1** [reply]
Sorry its not working for you. I've not had that bug before. It would be best to open a new issue to track it and also get some log data. If you add this to your configuratiin file and then restart:
 logger:
  default: critical
  logs:
    custom_components.battery_sim: debug

If you leave it to run for a few minutes go to logs then and click "load full log" you should see entries from the battery saying it's been set up and then each time it receives an update. Please include this log in the issue.
